gobblin-service/src/main/java/org/apache/gobblin/service/modules/spec/JobExecutionPlan.java

```diff
@@ -52,7 +54,7 @@
  * where the {@link JobSpec} will be executed.
  */
 @Data
-@EqualsAndHashCode(exclude = {"executionStatus", "currentAttempts"})
+@EqualsAndHashCode(exclude = {"executionStatus", "currentAttempts", "azkabanExecId"})
```

Comment:
JobExecutionPlan should have CompletableFuture instead of azkabanExecId. This way, the underlying SpecExecutor need not be exposed here.